File: app/src/main/java/cn/ifafu/ifafu/data/announce/SchoolCode.java
package cn.ifafu.ifafu.data.announce;


import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import cn.ifafu.ifafu.app.School;

@IntDef(value = {School.FAFU, School.FAFU_JS})
@Retention(RetentionPolicy.SOURCE)
public @interface SchoolCode {

}

File: app/src/main/java/cn/ifafu/ifafu/data/bean/SchoolCode.kt
package cn.ifafu.ifafu.data.bean

import androidx.annotation.StringDef
import cn.ifafu.ifafu.data.entity.User

@MustBeDocumented
@Target(
        AnnotationTarget.TYPE_PARAMETER,
        AnnotationTarget.FIELD
)
@StringDef(value = [User.FAFU, User.FAFU_JS])
annotation class SchoolCode